README.md
![Bağlama Rule!](https://codeberg.org/_aris/baglama/raw/branch/main/assets/banner-1544x500.png)

# Bağlama

Minimalist, elegant, usable publishing dispositive for WordPress. Still experimental and in developpement

### Description

- A clean, simplified and elegant WordPress admin design
- Primary focus on the content writing and editing experience 
- Remove unnecessary nags and bloat from core and some plugins

More information to come in the [Wiki](https://codeberg.org/_aris/baglama/wiki) pages on Codeberg.

### Compatibility

- For WordPress, this plugin is developed and tested with versions 6.0.* and 6.1. Warning: _not working_ with previous versions!
- For PHP, this plugin is developped with version 7.3.* and 7.4.* and compatibility is tested from 7.0.* up to 7.4.* but at this time _not_ realy with PHP 8.*
- Compatibility with plugins is still a _work in progress_. For now we tested compatibility in deep: Git Updater, Block Manager, Find my Blocks!, Code Snippets, Simply Static, Plugin Dependencies, Twentig, Custom Post type UI, WordPress Beta Tester...

### Privacy

This plugin does not collect or store any user data. It does not set any cookies, and it does not connect to any third-party locations. Thus, this plugin does not affect user privacy in any way.

### License

[GNU General Public License v2 or later](/_aris/baglama/src/branch/main/LICENSE)   
Official version of the GNU/GPL on the [GNU Project](http://www.gnu.org/licenses/gpl-2.0.html) Web site.   

### Links

- [Codeberg](https://codeberg.org/_aris/baglama/) (main repository)
- [tmprs.net](https://gitea.tmprs.net/tmprs/baglama) (mirror repository)
- [Github](https://github.com/bugsysop/baglama) (mirror repository)


CHANGES.md
### [Unreleased]

* Tweak the design of settings page (very minimal)
* Add server info (MySQL and PHP versions) to Admin Dashboard (At a Glance widget)
* When authors archive page is disabled: remove also authors from WP sitemap

### 0.2.3

Critical Bug
* Correction of a wrong module inclusion (this was killing WP when RSS setting where activated)

### 0.2.2

Functionalities
* Add a bulk plugins installation CLI tool (`baglama.sh`)
* Option to disable default WordPress block patterns

#### 0.2.1

Bugs and problems
* Fix `admin.css` (interference with Editor styles)

#### 0.2

Functionalities
* Minimalist user interface and default color scheme
* WP pages (About, Credits, Freedom, Privacy) under `Dashboard` main menu
* Extend the `sanitize` files name function
* Add access to Block Theme edition in `Admin Menu` (under `Apparence`)
* Add `Settings` page to manage optional modules
* Disable comments (option)
* Disable all RSS feeds (option)
* Disable authors archives page (option)

Bugs and problems
* Fix non working sanitize files name function
* Conditional code for `Admin Menu` tweaks
* Remove fat from custom admin CSS
* Better admin interface with CSS tweaks

#### 0.1.6

Functionalities
* Hide WordPress update nag to all but admins
* Hide Wordpress `PHP version` upgrade nag
* Hide some invasive plugins nags
* Disable administration email verification prompt
* Howdy? No more in `admin bar`
* Remove ajacent post prefetching (head)
* Add photo credit info to images
* Remove comments support and interface (inactive by default)
* Add a ID column for posts and pages
* Reusable Blocks: now in `submenu` under `Apparence`

Bugs and problems
* Remove access to themes and plugins files Edit for WP 6.1 

#### 0.1.5

Correcting the crash of version 0.1.4: parse error: two `<?php` tags in `engine/wordpress-config.php` file (see f0268f9b). Crappy code, sorry!

#### 0.1.4
Functionalities
* Redirect single user after successful login
* Add `Bye bye JQuery`(for frontend) 
* Remove useless options from user profile
* Remove the color picker from profile admin page
* Redirect user after successful login  

Enhancements
* Add a changelog: `CHANGES.md` (git-updater)

Bugs and problems
* Remove generator in head (was incomplete)
* Correct deprecated code for login page logo (#4)

#### 0.1.3

Functionalities
* Removing the "capital P dangit" function (suck)
* Remove DNS prefetch (privacy)
* Remove WP Emoji DNS prefetch (performances)

Bugs and problems
* Remove `generator` in head (was incomplete)

#### 0.1.2
_Test release for git-updater pluggin, no real changes_

Enhancements
* Add `git` support for updates

#### Initial commit

Functionalities
* Remove the admin bar in public site for all users
* Remove title prefix from archives pages
* Better RSS feeds (add thumbnail image and delay publishing)
* Clean and simple login page
* Stop WordPress to send too much mails
* Sanitize media name (all lower case)
* Disable attachement permalink
* Remove post format support (in case)
* Disable eimoji
* Prevent self ping
* No error message on login page
* Disable WP Registration Page
* No Password recovery link on login page
* No way to list user accounts from outside admin
* Clean-up site `head` from useless mentions

Bugs and problems
* None known ;-)


baglama.php
<?php
/*
Plugin Name: Bağlama
Plugin URI: https://codeberg.org/_aris/baglama
Description: Minimalist, elegant, usable publishing dispositive for WordPress. Still experimental and in developpement
Author: aris~
Text Domain: baglama
Author URI: https://papatheodorou.net/
Gitea Plugin URI: https://codeberg.org/_aris/baglama
Primary Branch: main
License: GNU General Public License v2 or later
License URI: http://www.gnu.org/licenses/gpl-2.0.txt
Version: 0.2.3
Requires at least: 6.0
Tested up to: 6.1
Requires PHP: 7.2
*/

// Exit if accessed directly.
if ( ! defined( 'ABSPATH' ) ) {
	exit;
}

// Settings
$baglama_plugin_version = '0.2.3';
require_once plugin_dir_path( __FILE__ ) . 'engine/settings-helpers.php';
require_once plugin_dir_path( __FILE__ ) . 'engine/settings-page.php';

// Text domain
load_plugin_textdomain( 'baglama', false, dirname( plugin_basename( __FILE__ ) ) . '/languages' );

// Engine

include_once plugin_dir_path( __FILE__ ).'engine/wordpress-config.php';
include_once plugin_dir_path( __FILE__ ).'engine/site-support.php';
if ( is_admin() ) {
    include_once plugin_dir_path( __FILE__ ).'engine/admin-mail-no-more.php';
    include_once plugin_dir_path( __FILE__ ).'engine/content-medias.php';
}
include_once plugin_dir_path( __FILE__ ).'engine/head-clean-up.php';
include_once plugin_dir_path( __FILE__ ).'engine/site-security-and-spam.php';

// Backoffice

if ( is_admin() ) {
    include_once plugin_dir_path( __FILE__ ).'backoffice/slate-functions.php';
    include_once plugin_dir_path( __FILE__ ).'backoffice/admin-menu.php';
    include_once plugin_dir_path( __FILE__ ).'backoffice/admin-hide-nags.php';
    include_once( plugin_dir_path( __FILE__ ).'backoffice/dashboard-widgets-tweaks.php' );
    include_once plugin_dir_path( __FILE__ ).'backoffice/admin-ui-tweaks.php';
    include_once plugin_dir_path( __FILE__ ).'backoffice/user-profile-tweaks.php';
    //include_once plugin_dir_path( __FILE__ ).'backoffice/widgets.php';
}

// Based on Slate Admin Theme - 1.2.4
add_action( 'admin_enqueue_scripts', 'baglama_admin_assets' );
function baglama_admin_assets() {
    wp_enqueue_style( 'baglama-admin-theme', plugins_url('assets/css/admin.css', __FILE__), array(), '1.2.4' );
    wp_enqueue_script( 'baglama', plugins_url( "assets/js/admin.js", __FILE__ ), array( 'jquery' ), '1.2.4' );
}
add_action( 'after_setup_theme', 'baglama_add_editor_styles' );
function baglama_add_editor_styles() {
    add_editor_style( plugins_url('css/editor-style.css', __FILE__ ) );
}

// Modules

include_once plugin_dir_path( __FILE__ ).'modules/admin-bar-no-more.php';
include_once plugin_dir_path( __FILE__ ).'modules/content-image-credit.php';
include_once plugin_dir_path( __FILE__ ).'modules/content-rss-feeds.php';
include_once plugin_dir_path( __FILE__ ).'modules/content-archives-pages.php';
include_once plugin_dir_path( __FILE__ ).'modules/site-login-interface.php';
/* if ( is_admin() ) {
include_once plugin_dir_path( __FILE__ ).'modules/site-no-comments.php';
} */

// Options

$baglama_tools_options = get_option('baglama_tools_option_name');
if ($baglama_tools_options !== null && $baglama_tools_options !== false) {
    if (baglama_check_valid_option("baglama_tools_comments_function_cbx1")) {
        // -- Comments
        require_once __DIR__ . '/modules/site-no-comments.php';
    }
    if (baglama_check_valid_option("baglama_tools_feeds_function_cbx2")) {
        // -- RSS Feeds
        require_once __DIR__ . '/modules/content-rss-feeds-no-links.php';
        require_once __DIR__ . '/modules/content-rss-feeds-disable.php';
    }
    if (baglama_check_valid_option("baglama_tools_authors_function_cbx3")) {
        // -- Author archives
        require_once __DIR__ . '/modules/content-author-archives.php';
    }
		if (baglama_check_valid_option("baglama_tools_patterns_function_cbx4")) {
        // -- Wordpress Core Patterns
        //require_once __DIR__ . '/modules/content-XXX.php';
				remove_theme_support( 'core-block-patterns' );
    }
} else {
    // -- Empty options
    $baglama_tools_options = array();
    $baglama_tools_options['baglama_plugin_version'] = $baglama_plugin_version;
    update_option('baglama_tools_option_name', $baglama_tools_options);
}

// WP Extensions page: add the settings and git links 
add_filter('plugin_action_links_' . plugin_basename(__FILE__), 'baglama_public_links_settings');
function baglama_public_links_settings($links)
{
    $new = array(
        'baglama-links-settings' => sprintf(
            '<a href="%s">%s</a>',
            esc_url(admin_url('options-general.php?page=baglama-tools')),
            esc_html__('Settings', 'baglama')
        ),
        'baglama-links-git' => sprintf(
            '<a href="%s" target="_blank">%s</a>',
            'https://codeberg.org/_aris/baglama',
            esc_html__('Code', 'baglama')
        ),
    );
    return array_merge($new, $links);
}
